packages/api/endpoints/granules.js

```diff
@@ -678,6 +706,115 @@ const associateExecution = async (req, res) => {
   });
 };
 
+const BulkPatchGranuleCollectionSchema = z.object({
+  apiGranules: z.array(z.object({}).catchall(z.any())).nonempty(),
+  collectionId: z.string().nonempty(),
+  esConcurrency: z.number().positive(),
+}).catchall(z.unknown());
+
+const BulkPatchSchema = z.object({
+  apiGranules: z.array(z.object({}).catchall(z.any())).nonempty(),
+  dbConcurrency: z.number().positive(),
+  dbMaxPool: z.number().positive(),
+}).catchall(z.unknown());
+
+const parseBulkPatchGranuleCollectionPayload = zodParser('BulkPatchGranuleCollection payload', BulkPatchGranuleCollectionSchema);
+const parseBulkPatchPayload = zodParser('BulkPatchSchema payload', BulkPatchSchema);
+
+/**
+ * Update a batch of granules to change collectionId to a new collectionId
+ * in PG and ES
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function bulkPatchGranuleCollection(req, res) {
+  const {
+    collectionPgModel = new CollectionPgModel(),
+    knex = await getKnexClient(),
+    esClient = await getEsClient(),
+    mappingFunction = pMap,
+  } = req.testContext || {};
+  req.body.esConcurrency = req.body.esConcurrency ?? 5;
+  const body = parseBulkPatchGranuleCollectionPayload(req.body);
+  if (isError(body)) {
+    return returnCustomValidationErrors(res, body);
+  }
+
+  const granules = req.body.apiGranules;
+  const granuleIds = granules.map((granule) => granule.granuleId);
+  const newCollectionId = req.body.collectionId;
+  const collection = await collectionPgModel.get(
+    knex,
+    deconstructCollectionId(newCollectionId)
+  );
+
+  try {
+    await mappingFunction(
```

Comment:
One last comment/question/clarification

In this setup, we're mapping over a batch of updates.  If one fails, we get no feedback from the endpoint re: the state of individual granules, just <failed batch>.  

Does that work for the upstream branch?       Assuming so, I think we should document that behavior (users may want to not use this without changes) and probably write a ticket to update the endpoint to return success/failed { graunleId, collectionId } arrays via Promise.allSettled (or the sindre implementation of it at least). 

Thoughts, particularly @etcart ?

Reply:
Discussed this 'in person', documentation is sufficient at this time per @etcart